src/megachatapi_impl.cpp

```diff
@@ -1806,6 +1806,16 @@ int MegaChatApiImpl::init(const char *sid)
     return MegaChatApiImpl::convertInitState(state);
 }
 
+void MegaChatApiImpl::resetClientid()
+{
+    sdkMutex.lock();
+    if (mClient)
+    {
+        mClient->resetClientid();
```

Comment:
Perhaps an `assert()` to ensure the initialization state is still waiting-new-session or has-offline-session, but not higher. Otherwise, it means the developer is calling this function too late.